source/core/src/main/com/csse3200/game/components/station/StationItemHandlerComponent.java

```diff
@@ -0,0 +1,105 @@
+package com.csse3200.game.components.station;
+
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import com.csse3200.game.components.Component;
+
+public class StationItemHandlerComponent extends Component {
+    /**
+     * String type - storing type of station
+     * StationInventoryComponent inventorycomponent - instance of inventory for this station
+     * TBD acceptableItems - HashMap, HashSet etc of mappings for acceptable items based on station
+     */
+    private final String type;
+    private StationInventoryComponent inventoryComponent;
+    private HashSet<String> acceptableItems = new HashSet<>();
+
+    // General TODO:
+    // Add trigger calls to external for failed interactions
+    // Introduce an actual structure for acceptable items, json parsing etc
+    // Processing in Inventory component, animation, timing and mapping
+    // Create subclass for each station where needed, eg classic bench will need
+    //      to call add second component method that we dont want all stations to be able to access
+
+    /**
+     * General constructor
+     * @param type - storing type of station
+     * @param acceptableItems - HashMap, HashSet etc of mappings for acceptable items based on station
+     */
+    public StationItemHandlerComponent(String type, ArrayList<String> acceptableItems) {
+        this.type = type;
+    }
+
+    /**
+     *  Called on creation of the station to allow outside interaction within the station.
+     *  Adds the listener for set current item and for remove current item.
+     */
+    @Override
+    public void create() {
+        acceptableItems.add("meat");
+        acceptableItems.add("vegetable");
```

Comment:
Why adding the acceptable items already?

Reply:
For testing of data structure, should have been an included comment will add